services/motion/explore/explore.go

```diff
@@ -0,0 +1,626 @@
+	resource.Actuator
+	referenceframe.InputEnabled
+}
+
+// obstacleDetectorName is used for sending and receiving all obstacle detectors until the motionCfg can be
+// added to the move service.
+type obstacleDetectorName struct {
+	VisionService resource.Name
+	Camera        resource.Name
+}
+
+// obstacleDetectorPair provides a map for matching vision services to any and all cameras to be used with them.
+type obstacleDetectorPair map[vision.Service][]camera.Camera
+
+// ErrNotImplemented is thrown when an unreleased function is called.
+var ErrNotImplemented = errors.New("function coming soon but not yet implemented")
+
+// Config describes how to configure the service; currently only used for specifying dependency on frame
+// system service.
+type Config struct {
+	LogFilePath string `json:"log_file_path"`
+}
+
+// Validate here adds a dependency on the internal framesystem service.
+func (c *Config) Validate(path string) ([]string, error) {
+	return []string{framesystem.InternalServiceName.String()}, nil
+}
+
+// NewExplore returns a new explore motion service for the given robot.
+func NewExplore(ctx context.Context, deps resource.Dependencies, conf resource.Config, logger golog.Logger) (motion.Service, error) {
+	ms := &explore{
+		Named:                 conf.ResourceName().AsNamed(),
+		logger:                logger,
+		obstacleResponseChan:  make(chan moveResponse),
+		executionResponseChan: make(chan moveResponse),
+		backgroundWorkers:     &sync.WaitGroup{},
+	}
+
+	if err := ms.Reconfigure(ctx, deps, conf); err != nil {
+		return nil, err
+	}
+	return ms, nil
+}
+
+// Reconfigure updates the explore motion service when the config has changed.
+func (ms *explore) Reconfigure(
+	ctx context.Context,
+	deps resource.Dependencies,
+	conf resource.Config,
+) (err error) {
+	ms.lock.Lock()
+	defer ms.lock.Unlock()
+
+	config, err := resource.NativeConfig[*Config](conf)
+	if err != nil {
+		return err
+	}
+	if config.LogFilePath != "" {
+		logger, err := utils.NewFilePathDebugLogger(config.LogFilePath, "motion")
+		if err != nil {
+			return err
+		}
+		ms.logger = logger
+	}
+
+	// Iterate over dependence is and store components and services along with the frame service directly
+	components := make(map[resource.Name]resource.Resource)
+	services := make(map[resource.Name]resource.Resource)
+	for name, dep := range deps {
+		switch dep := dep.(type) {
+		case framesystem.Service:
+			ms.fsService = dep
+		case vision.Service:
+			services[name] = dep
+		default:
+			components[name] = dep
+		}
+	}
+
+	ms.components = components
+	ms.services = services
+	return nil
+}
+
+type explore struct {
+	resource.Named
+	resource.TriviallyCloseable
+	frameSystem referenceframe.FrameSystem
+	fsService   framesystem.Service
+	components  map[resource.Name]resource.Resource
+	services    map[resource.Name]resource.Resource
+	logger      golog.Logger
+	lock        sync.Mutex
+
+	obstacleResponseChan  chan moveResponse
```

Comment:
[note] after full integration into BuiltIn these will probably be moved or accessed via the moveRequest/moveAttempt files and will disappear for the general struct.